util/tracing/tracer.go

```diff
@@ -140,18 +140,52 @@ func ChildSpan(ctx context.Context, opName string) (context.Context, func()) {
 	return opentracing.ContextWithSpan(ctx, newSpan), func() { newSpan.Finish() }
 }
 
+// netTraceIntegrator is passed into basictracer as NewSpanEventListener
+// and causes all traces to be registered with the net/trace endpoint.
+func netTraceIntegrator() func(basictracer.SpanEvent) {
+	var tr trace.Trace
+	return func(e basictracer.SpanEvent) {
+		switch t := e.(type) {
+		case basictracer.EventCreate:
+			tr = trace.New("tracing", t.OperationName)
+			// TODO(radu): use maxLogsPerSpan here when #9748 is fixed.
+			tr.SetMaxEvents(20)
```

Comment:
Calling `SetMaxEvents` at all is notably worse than using the default:

```
  60.0s: 293813 total 4896.6/sec
```

vs (`tr.SetMaxEvents(10)`)

```
  60.0s: 294791 total 4913.1/sec
```

vs (not calling `SetMaxEvents` at all)

```
  60.0s: 303401 total 5056.6/sec
```